tests/test_tables.py

```diff
@@ -0,0 +1,206 @@
+from .base import BaseTinyFatTest
+from .utils import match_all_elements
+from tinyfatdb import TinyFatDB, TinyFatTable
+from tinydb import TinyDB
+from tinydb import Query as Q
+
+a_ends_with_2 = Q().a.test(lambda a: a.endswith("2"))
+
+
+########################################################################
+class TableAndModelTestSuite:
+    _entries = ({"a": "A1", "b": "B1", "c": "C1"},
+                {"a": "A2", "b": "B2", "c": "C2"},
+                {"a": "A3", "b": "B3", "c": "C3"})
+
+    ####################################################################
+    def test_purge_tables(self):
+        self.db.table("Foo", table=TinyFatTable)
+        self.db.purge_tables()
+        self.assertEqual(set(), self.db.tables())
+
+    ####################################################################
+    def test_insert(self):
+        self.assertEqual(len(self.entries), len(self.db))
+        self.db.insert({})
+        self.assertEqual(len(self.entries) + 1, len(self.db))
+
+    ####################################################################
+    def test_insert_multiple(self):
+        self.assertEqual(len(self.entries), len(self.db))
+        self.db.insert_multiple(({}, {}))
+        self.assertEqual(len(self.entries) + 2, len(self.db))
+
+    ####################################################################
+    def test_all(self):
+        self.assertEqual(self.entries, tuple(self.db.all()))
+
+    ####################################################################
+    def test_get_by_eid(self):
+        entry = self.entries[0]
+        self.assertEqual(entry, self.db.get(eid=entry["eid"]))
+
+    ####################################################################
+    def test_get_by_condition(self):
+        self.assertEqual(self.entries[1], self.db.get(a_ends_with_2))
+
+    ####################################################################
+    def test_count(self):
+        self.assertEqual(1, self.db.count(Q().a == "A1"))
+        self.assertEqual(3, self.db.count(Q().a.test(lambda val: "A" in val)))
+
+    ####################################################################
+    def test_contains_by_eid(self):
+        good_eid = self.entries[0]["eid"]
+        bad_eid = 72
```

Comment:
This is minor, but it would probably be better to use a number that will never possibly be returned as an`eid`, such as `-99`